src/util/bip32.rs

```diff
@@ -498,11 +499,26 @@ impl ExtendedPrivKey {
             depth: 0,
             parent_fingerprint: Default::default(),
             child_number: ChildNumber::from_normal_idx(0)?,
-            private_key: PrivateKey::from_slice(&hmac_result[..32], network)?,
+            private_key: secp256k1::SecretKey::from_slice(&hmac_result[..32])?,
             chain_code: ChainCode::from(&hmac_result[32..]),
         })
     }
 
+    /// Constructs ECDSA compressed private key matching internal secret key representation.
+    pub fn to_ecdsa_priv(&self) -> ecdsa::PrivateKey {
```

Comment:
I wonder if the `_priv` part is really useful. You know you are operating on an xpriv and the return type is a private key. Kinda related: I'd like to explore the possibility of putting the derive and conversion functions into a trait (not here though) to allow generic code over x-keys.

Reply:
After #755, I think this should be just `to_priv`. As mentioned and discussed in several places, keys are not related to signature algorithms.